Useful tools & utilities/Libraries.md

```diff
@@ -106,11 +94,33 @@
 
 	More info with code examples is available [here](https://github.com/JakeWharton/ThreeTenABP).
 
-3. **Clean status bar**
+3. **Dagger/Hilt**
+
+    Dagger is a dependency injection library for Android, and it is a must-have on every medium to large Android project. It creates global and scoped graphs of depedencies and provides them where needed through injection.
+
+    Hilt is a library built on top of Dagger to simplify the setup process.
+
+    More info on Hilt (recommended) with code examples is available [here](https://developer.android.com/training/dependency-injection/hilt-android).
+
+    For Dagger without Hilt, info and code examples are available [here](https://developer.android.com/training/dependency-injection/dagger-android).
+
+4. **Android KTX**
+
+    Android KTX is a set of Kotlin extensions that are included with Android Jetpack and other Android libraries. KTX extensions provide idiomatic Kotlin to Jetpack, Android platform, and other APIs
+
+    Every project written in Kotlin should use these as they make your code more concise, readable and shorter.
+
+    More info with code examples is available [here](https://developer.android.com/kotlin/ktx).
+
+5. **Lifecycle**
+
+    Lifecycle from AndroidX includes everything we need to make a good MVVM application such as: ViewModels, LiveData, Lifecycle callbacks etc.
+
+    It is essential for MVVM arcitecture, but it can also help in lost of other use cases.
```

Comment:
I'm not sure what you meant by "help in lost of other use cases", could you explain? 😄